cdap-app-fabric/src/main/java/io/cdap/cdap/security/impersonation/DefaultUGIProvider.java

```diff
@@ -172,4 +204,53 @@ private File localizeKeytab(Location keytabLocation) throws IOException {
 
     return localKeytabFile.toFile();
   }
+
+  /**
+   * Returns the runtime user supplied arguments to a program if entity is of type ProgramRun
+   *
+   * @param programId the entity id
+   * @return properties map
+   */
+  private Map<String, String> getRuntimeProperties(NamespacedEntityId entityId) {
+    if (!(entityId instanceof ProgramRunId)) {
+      LOG.debug("Entity Id not of type ProgramRunId, skipping checking of runtime args");
+      return Collections.emptyMap();
+    }
+
+    ProgramRunId runId = ((ProgramRunId) entityId);
+    RunRecordMeta runRecord = null;
+
+    long sleepDelayMs = TimeUnit.MILLISECONDS.toMillis(100);
+    long timeoutMs = TimeUnit.SECONDS.toMillis(10);
+
+    try {
+      runRecord = Retries.callWithRetries(() -> {
+        RunRecordMeta rec = store.getRun(runId);
+        if (rec != null) {
+          return rec;
+        }
+        throw new Exception("Retrying again to fetch run record");
+      }, RetryStrategies.timeLimit(timeoutMs, TimeUnit.MILLISECONDS,
+                  RetryStrategies.fixDelay(sleepDelayMs, TimeUnit.MILLISECONDS)),
+                  Exception.class::isInstance);
+    } catch (Exception e1) {
+      LOG.debug("Exception while trying to fecth RunRecord %s ", e1.getMessage());
+    }
+
+    if (runRecord == null) {
```

Comment:
You don't need this check as it can be done and return from the `catch` block above.

```java
try {
  Retries.callWithRetries(...);
} catch (Exception e) {
  LOG.warn("Failed to fetch run record for {} due to {}", runId, e.getMessage(), e);
  return Collections.emptyMap();
}
```

Reply:
Ok. Removed now.